packages/select/test/multiSelectTests.tsx

```diff
@@ -119,37 +123,44 @@ describe("<MultiSelect>", () => {
         assert.strictEqual(wrapper.find(Popover).prop("isOpen"), false);
         findTargetButton(wrapper).simulate("click");
 
-        // MultiSelect uses requestAnimationFrame for the PopoverInteraction causing a slight delay
-        await delay(20);
-        wrapper.update();
-
         assert.strictEqual(wrapper.find(Popover).prop("isOpen"), true);
     });
 
     it("allows searching within popover content when custom target provided", async () => {
+        // Mount to document for this test to check from input focus
+        const containerElement = document.createElement("div");
+        document.body.appendChild(containerElement);
+
         const customTarget = <Button data-testid="custom-target-button" text="Target" />;
         const handleQueryChange = sinon.spy();
-        const wrapper = multiselect({
+        const props = {
             customTarget,
             onQueryChange: handleQueryChange,
             popoverProps: { usePortal: false },
+        };
+
+        const wrapper = mount(<MultiSelect<Film> {...defaultProps} {...handlers} {...props} />, {
+            attachTo: containerElement,
         });
 
         findTargetButton(wrapper).simulate("click");
 
-        // MultiSelect uses requestAnimationFrame for the PopoverInteraction causing a slight delay
-        await delay(20);
+        // There's a slight delay between the Popover rendering and input getting focus
```

Comment:
This can be visible in actual use, and looks like Select suffers from this delay as well. My guess its due to ref callbacks taking a second after mounting so it can't call focus right away

Reply:
Wasn't able to fully get down to what causes the delay (spent a decent amount digging around) best guess is that ref hypothesis?

Don't think requestAnimationFrame is the culprit though: Playing around with the Select component and the delay on the input focus, it's the same whether the `focus()` is inside or outside of the requestAnimationFrame callback

https://github.com/user-attachments/assets/b6f7b535-a7b6-48e5-970d-8877c5382c2b